code/modules/events/queen_mothers_blessing.dm

```diff
@@ -0,0 +1,37 @@
+#define QM_HEAL_AMOUNT 250
+
+/datum/round_event_control/queen_mothers_blessing
+	name = "Queen Mother's Blessing"
+	typepath = /datum/round_event/queen_mothers_blessing
+	weight = 15
+	earliest_start = 10 MINUTES
+
+	gamemode_blacklist = list("Combat Patrol","Civil War","Sensor Capture")
+
+/datum/round_event_control/queen_mothers_blessing/can_spawn_event(players_amt, gamemode)
+	if(length(GLOB.alive_xeno_list) <= 4)
+		return FALSE
+	return ..()
+
+/datum/round_event/queen_mothers_blessing/start()
+	var/sound/queen_sound = sound(get_sfx("queen"), channel = CHANNEL_ANNOUNCEMENTS)
+	for(var/mob/living/carbon/xenomorph/receiving_xeno in GLOB.alive_xeno_list)
+		SEND_SOUND(receiving_xeno, queen_sound)
+	for(var/mob/living/carbon/xenomorph/X in shuffle(GLOB.alive_xeno_list))
+		if(isxenolarva(X))
+			continue
+		else if(isminion(X))
+			continue
+		bless_xeno(X)
+		break
```

Comment:
It probably doesn't matter, but I would personally have this as a return, as it's more consistant/clear that there's nothing left in the proc to do.